### PR TITLE
Document Autotools and CMake configure options

### PR DESCRIPTION
**Pull request by mwtoews**
This is a draft document, as I'd like to know if there are any other build configure options that are worth documenting. Drop a comment here, and I'll add it in, otherwise I'll commit this in a week or so.

### CODE REVIEW

File: docs/source/install.rst [hunk 1]
```diff
@@ -200,6 +200,47 @@ The test suite requires that the proj-datumgrid package is installed in
...
+
+    C++ compiler flags
+
+See ``./configure --help`` for all options, here are a few key options:
+
+.. option:: --enable-lto
+
+    Enable compiler's Link Time Optimization, default disabled.
+
+.. option:: --disable-tiff
```
Comment (kbevers) [inline docs/source/install.rst:235]: Should also include `--with-curl`  here

**Review by kbevers — APPROVED** [review]
Looks good to me, thanks Mike

File: docs/source/install.rst (hunk 1 above)
Comment (mwtoews) [reply docs/source/install.rst:235]: Added, along with CMake's `CURL_INCLUDE_DIR` and `CURL_LIBRARY` options.

File: docs/source/install.rst [hunk 2]
```diff
@@ -200,6 +200,51 @@ The test suite requires that the proj-datumgrid package is installed in
 :envvar:`PROJ_LIB`.
 
 
+Autotoos configure options
```
Comment (rouault) [inline docs/source/install.rst]:
```suggestion
Autotools configure options
```

File: docs/source/install.rst [hunk 3]
```diff
@@ -240,10 +285,96 @@ Tests are run with::
...
+    Path to an ``sqlite3`` or ``sqlite3.exe`` executable.
+
+.. option:: SQLITE3_INCLUDE_DIR
+
+    Path to an include directory with the ``sqlite3.h`` header file.
+
+.. option:: SQLITE3_LIBRARY
 
+    Path to a shared or static library file, such as ``sqlite3.dll``,
+    ``libsqlite3.so``, ``sqlite3.lib`` or other name.
```
Comment (rouault) [inline docs/source/install.rst:398]: for consistency TIFF_INCLUDE_DIR and TIFF_LIBRARY should also likely be documented.

File: docs/source/install.rst [hunk 4]
```diff
@@ -200,6 +200,51 @@ The test suite requires that the proj-datumgrid package is installed in
...
+
+See ``./configure --help`` for all options, here are a few key options:
+
+.. option:: --enable-lto
+
+    Enable compiler's Link Time Optimization, default disabled.
+
+.. option:: --disable-tiff
+
+    TIFF support is enabled by default to read some grids, but this can
```
Comment (rouault) [inline docs/source/install.rst]: Perhaps a note should be added that if PROJ is used with the proj-data package, which is the nominal use case, TIFF support must be enabled.

File: docs/source/install.rst (hunk 3 above)
Comment (mwtoews) [reply docs/source/install.rst:398]: oddly there were `TIFF_LIBRARY_RELEASE` and `TIFF_LIBRARY_DEBUG` variables

File: docs/source/install.rst (hunk 4 above)
Comment (mwtoews) [reply docs/source/install.rst]: I've reworded both places. Are we using proj-data or PROJ-data? (i.e. the package name or the GitHub repo name?)

**Review by rouault — APPROVED** [review]

**Pull request merged**